File: ec2/server/src/entity/feedback.ts
import {
  Entity,
  PrimaryGeneratedColumn,
  Column,
  ManyToOne,
  BaseEntity,
  JoinColumn,
} from "typeorm";
import { Target } from "./target";

@Entity()
export class Feedback extends BaseEntity {
  @PrimaryGeneratedColumn()
  id: number;

  @Column({ length: 100 })
  name: string;

  @Column()
  message: string;

  @Column()
  password: string;

  @Column({ nullable: true })
  image_url?: string;

  @Column({ type: "int" })
  targetId: number;

  @ManyToOne(() => Target, (target) => target.donations)
  @JoinColumn({ name: "targetId" })
  target: Target;
}

File: app/lib/domain/feedback/feedback.dart
class Feedback {
  final int? id;
  final String? name;
  final String? message;
  final int? targetId;
  final String? imageUrl;
  Feedback({
    this.id,
    this.name,
    this.message,
    this.targetId,
    this.imageUrl,
  });

  // 통신을 위해서 json 처럼 생긴 문자열 {"id":1} => Dart 오브젝트

  Feedback.fromJson(Map<String, dynamic> json)
      : id = json["id"],
        name = json["name"],
        message = json["message"],
        targetId = json["targetId"],
        imageUrl = json["image_url"];

  bool hasImage() => imageUrl != null;
}
